Add Discord Rich Presence

FleetAdmiralButter opened the pull request:
Changed:

1) Adds support for Discord Rich Presence integration:

![image](https://user-images.githubusercontent.com/37036084/147652767-1c853c70-1879-4651-98fa-45faebffcac3.png)

marzent commented:
Hm 
This is actually a very nice feature, however I was also planning on porting https://github.com/0e4ef622/wine-discord-ipc-bridge to macOS in the near future, especially since we are basically only wrapping xivlauncher which already has quite a nice discord rich presence as well….

I’m kinda conflicted on this tbh, as long as it doesn’t really add much to what discord is showing when you manually add the game it feels kinda like advertising ;)

What do you think?

FleetAdmiralButter commented:
@marzent

I added some notes in Discord but will put a more structured form here.

> it doesn’t really add much to what discord is showing

It does show a good deal more than manually adding the .app to Discord. Most notably it includes the icon and playtime.

> it feels kinda like advertising

Agree with you here.

How about this:

![image](https://user-images.githubusercontent.com/37036084/147719949-188ef970-f53d-44e8-8acc-20a56a8dcce2.png)

Cut down on the info shown in the flyout. Just display the app name, logo, and playtime.

We could take it even further and even add an option to disable this outright if someone doesn't want it:
![image](https://user-images.githubusercontent.com/37036084/147719976-8852c879-dd2c-4f23-9b64-9d2fa99026b6.png)


> I was also planning on porting https://github.com/0e4ef622/wine-discord-ipc-bridge to macOS in the near future

I'm thinking this could be a stopgap between now and when that library gets ported over. I'm not averse to removing this implementation once we have something better.

FleetAdmiralButter commented:
@marzent

As discussed earlier I've added the toggle and simplified the flyout.

Toggle:
![image](https://user-images.githubusercontent.com/37036084/147732123-8ca4b562-dc7e-45de-b925-868602d5c409.png)

Flyout:
![image](https://user-images.githubusercontent.com/37036084/147732147-49f975ba-5bab-40c1-a681-bb2ee28c988c.png)
